run_test.sh

```diff
@@ -230,29 +230,30 @@ run_with_expected_error() {
     local output_log=test_cmd_output.log
     local expected_output_log=test_cmd_expected_output.log
     local error_code=0
+    local ret=0
 
     shift
     cat > ${expected_output_log}
     "$@" > ${stdout_log} 2> ${stderr_log} || error_code=$?
     cat ${stdout_log} ${stderr_log} > ${output_log}
     rm ${stdout_log} ${stderr_log}
-    if [[ "${error_code}" = "0" ]] ; then
-        echo "Error: command '$*' was successful"
-        cat ${output_log}
-        rm ${expected_output_log} ${output_log}
-        return 1
-    elif [[ "${error_code}" != "${expected_error_code}" ]] ; then
```

Comment:
AFAICT this was inspired by my request to unify with `run_with_expected_output()`, right? Thank you, for which

The new code-flow is ever so slightly more convoluted to follow. So how about this alternative:

 - drop the above if block -  we don't care if command is successful or not as long as it matches the expectation
   - update the expectations as needed - single instance it seems
 - live with the duplicate `rm ${expected_output_log} ${output_log}`

Reply:
right, there is no point in having three ways saying "this is not the expected result"